File: ios/common/SensorDataStorage.swift
//
//  SensorDataStorage.swift
//  SampleApp
//
//  Created by Zeki on 11.10.22.
//

import Foundation
class SensorDataStorage: Codable{
  static var sharedInstance = SensorDataStorage()

  var Data = [SensorsSample]()
  //var Data = SynchronizedArray<SensorsSample>()
  var Time:Double=0
  var TripId:Double=0
  var Tbt:Int = 0
}

File: android/app/src/main/java/sensor/SensorDataStorage.java
package sensor;

import java.util.ArrayList;

public class SensorDataStorage {
    private static SensorDataStorage instance = new SensorDataStorage();
    public ArrayList<SensorsSample> Data = new ArrayList<SensorsSample>();
    private long Time;
    private long TripId;
    private long Tbt;
    private long Id;

    public static SensorDataStorage getInstance()
    {
        if (instance == null)
            instance = new SensorDataStorage();

        return instance;
    }


    public long getTime() {
        return Time;
    }

    public void setTime(long time) {
        Time = time;
    }

    public long getTripId() {
        return TripId;
    }

    public void setTripId(long tripId) {
        TripId = tripId;
    }

    public long getTbt() {
        return Tbt;
    }

    public void setTbt(long tbt) {
        Tbt = tbt;
    }

    public long getId() {
        return Id;
    }

    public void setId(long id) {
        Id = id;
    }
}
